common/web/types/src/ldml-keyboard/ldml-keyboard-xml.ts

```diff
@@ -193,3 +194,36 @@ export interface LKDisplays {
   display?: LKDisplay[];
   displayOptions?: LKDisplayOptions;
 };
+
+export interface LKForms {
+  form?: LKForm[];
+};
+
+export interface LKForm {
+  id?: string;
+  scanCodes?: LKScanCodes[];
+};
+
+export interface LKScanCodes {
+  codes?: string;
+};
+
+/**
+ * Utilities for determining the import status of items
+ */
+export class ImportStatus {
+  /** item came in via implied (spec based) import, such as keys-Latn-implied.xml */
+  static impliedImport = Symbol.for('@keymanapp:implied_import');
+  /** item came in via import */
+  static import = Symbol.for('@keymanapp:import');
```

Comment:
This introduces a new namespace that we need to be careful with -- the `@keymanapp` Symbol namespace. Conceptually I think it is fine but I am just a little wary about future trouble with it. Might be good to use a longer prefix, for disambiguation e.g.:

```suggestion
  static impliedImport = Symbol.for('@keymanapp:common/ldml/implied_import');
  /** item came in via import */
  static import = Symbol.for('@keymanapp:common/ldml/import');
```

Also, just verifying is this just a way of tagging arbitrary structures with an "imported" flag? And you wanted to use this to avoid potential stumbling blocks in the way we use those structures, because it won't be an interable key on an object?

Reply:
On reflection, I think this doesn't need a named symbol at all. this symbol should be used only within one run of kmc.  it doesn't need to be independently detected. I will drop the namespace.

yes, it's just a way to tag the structure as imported - it's not iterable on the object, it doesn't affect the API at all at any level. It's metadata answering the question "how did this data object come in?"

Reply:
OK here?